tl

```diff
@@ -107,70 +113,55 @@ local function report_type_errors(result)
    end
 end
 
-local function die(msg)
-   printerr(msg)
-   os.exit(1)
-end
-
-local result, err = tl.process(filename, nil, nil, nil, tlconfig["preload_modules"])
-if err then
-   die(err)
-end
-
-local has_syntax_errors = report_errors("syntax error", result.syntax_errors)
-if has_syntax_errors then
-   os.exit(1)
-end
-
-local lua_name = filename:gsub(".tl$", ".lua")
-
-if cmd == "run" then
+for _, filename in ipairs(args["script"]) do
+   local result, err = tl.process(filename, nil, nil, nil, tlconfig["preload_modules"])
+   if err then
+      die(err)
+   end
 
-   if filename:match("%.tl$") then
-      report_type_errors(result)
+   local has_syntax_errors = report_errors("syntax error", result.syntax_errors)
+   if has_syntax_errors then
+      os.exit(1)
    end
 
-   local chunk = load(tl.pretty_print_ast(result.ast), "@" .. filename)
+   local lua_name = filename:gsub(".tl$", ".lua")
 
-   local narg = #arg
-   for i = -5, narg do
-      arg[i-2] = arg[i]
-   end
-   arg[narg] = nil
-   arg[narg - 1] = nil
+   if cmd == "run" then
+      if filename:match("%.tl$") then
+         report_type_errors(result)
+      end
 
-   return chunk()
+      local chunk = load(tl.pretty_print_ast(result.ast), "@" .. filename)
 
-elseif cmd == "check" then
+      local narg = #arg
+      for i = -5, narg do
+         arg[i - 2] = arg[i]
+      end
+      arg[narg] = nil
+      arg[narg - 1] = nil
 
-   report_type_errors(result)
+      return chunk()
+   elseif cmd == "check" then
+      report_type_errors(result)
 
-   print("========================================")
-   print("Type checked " .. filename)
-   print("0 errors detected -- you can use:")
-   print()
-   print("   tl run " .. filename )
-   print()
-   print("       to run " .. filename .. " as a program")
-   print()
-   print("   tl gen " .. filename)
-   print()
-   print("       to generate " .. lua_name)
-   os.exit(0)
+      print("========================================")
+      print("Type checked " .. filename)
+      print("0 errors detected")
+      print()
```

Comment:
I know these instructions you removed are sort of ugly, but I think at this early stage it's good to intstruct newcomers to ease the learning curve.

Reply:
OK, I brought them back.

Reply:
Thanks!